ding/reward_model/curiosity_reward_model.py

```diff
@@ -0,0 +1,201 @@
+        self.residual = [
+            nn.Sequential(
+                nn.Linear(action_shape + 512, 512),
+                nn.LeakyReLU(),
+                nn.Linear(512, 512),
+            ) for _ in range(8)
+        ]
+        self.forward_net_1 = nn.Sequential(nn.Linear(action_shape + feature_output, 512), nn.LeakyReLU())
+        self.forward_net_2 = nn.Sequential(nn.Linear(action_shape + 512, feature_output), )
+
+    def forward(self, state: torch.Tensor, next_state: torch.Tensor,
+                action_long: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+        r"""
+        Overview:
+            Use observation, next_observation and action to genearte ICM module
+            Parameter updates with ICMNetwork forward setup.
+        Arguments:
+            - state (:obj:`torch.Tensor`):
+                The current state batch, (B,N=observation_size)``.
+            - next_state (:obj:`torch.Tensor`):
+                The next state batch, (B,N=observation_size)``.
+            - action_long (:obj:`torch.Tensor`):
+                The action, (B,M=action_size)``.
+        Returns:
+            - real_next_state_feature (:obj:`torch.Tensor`):
+                Run with the encoder. Return the real next_state's embedded feature.
+            - pred_next_state_feature (:obj:`torch.Tensor`):
+                Run with the encoder and residual network. Return the predicted next_state's embedded feature.
+            - pred_action_logit (:obj:`Dict`):
+                Run with the encoder. Return the predicted action logit.
+        """
+        state = torch.FloatTensor(state)
+        next_state = torch.FloatTensor(next_state)
+        action_long = torch.LongTensor(action_long)
+        action = one_hot(action_long, num=self.action_shape).squeeze(1)
+        encode_state = self.feature(state)
+        encode_next_state = self.feature(next_state)
+        # get pred action logit
+        pred_action_logit = torch.cat((encode_state, encode_next_state), 1)
+        pred_action_logit = self.inverse_net(pred_action_logit)
+        # ---------------------
+
+        # get pred next state
+        pred_next_state_feature_orig = torch.cat((encode_state, action), 1)
+        pred_next_state_feature_orig = self.forward_net_1(pred_next_state_feature_orig)
+
+        # residual
+        for i in range(4):
+            pred_next_state_feature = self.residual[i * 2](torch.cat((pred_next_state_feature_orig, action), 1))
+            pred_next_state_feature_orig = self.residual[i * 2 + 1](
+                torch.cat((pred_next_state_feature, action), 1)
+            ) + pred_next_state_feature_orig
+        pred_next_state_feature = self.forward_net_2(torch.cat((pred_next_state_feature_orig, action), 1))
+        real_next_state_feature = encode_next_state
+        return real_next_state_feature, pred_next_state_feature, pred_action_logit
+
+
+@REWARD_MODEL_REGISTRY.register('curiosity')
+class ICMRewardModel(BaseRewardModel):
```

Comment:
add comments like this [file](https://github.com/opendilab/DI-engine/blob/main/ding/reward_model/gail_irl_model.py#L108) and config comments like [this](https://github.com/opendilab/DI-engine/blob/main/ding/policy/acer.py#L57)